### PR TITLE
Support nodeSelector

### PR DESCRIPTION
**Pull request by clyang82**
/assign @metalmatze

### CODE REVIEW

File: api/v1alpha1/observatorium_types.go [hunk 1]
```diff
@@ -45,6 +45,9 @@ type ObservatoriumSpec struct {
 	// Loki
 	// +optional
 	Loki *LokiSpec `json:"loki,omitempty"`
+	// NodeSelector applies to all components
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
```
Comment (metalmatze) [inline api/v1alpha1/observatorium_types.go:50]: Is this the data type used in Kubernetes too?

Comment (clyang82) [reply api/v1alpha1/observatorium_types.go:50]: Yes. see here https://pkg.go.dev/k8s.io/api@v0.18.3/core/v1#PodSpec

File: api/v1alpha1/observatorium_types.go [hunk 2]
```diff
@@ -45,6 +45,9 @@ type ObservatoriumSpec struct {
 	// Loki
 	// +optional
 	Loki *LokiSpec `json:"loki,omitempty"`
+	// NodeSelector applies to all components
```
Comment (squat) [inline api/v1alpha1/observatorium_types.go:48]:
```suggestion
	// NodeSelector causes all components to be scheduled on nodes with matching labels.
```

**Review by squat — APPROVED** [review]
LGTM one small suggestion for improving the documentation

**Review by squat — APPROVED** [review]

**Pull request merged**